flang/include/flang/Lower/IntrinsicCall.h

```diff
@@ -73,7 +73,7 @@ struct IntrinsicArgumentLoweringRules;
 /// Return argument lowering rules for an intrinsic.
 /// Returns a nullptr if all the intrinsic arguments should be lowered by value.
 const IntrinsicArgumentLoweringRules *
-getIntrinsicArgumentLowering(llvm::StringRef intrinsicName);
+getIntrinsicArgumentLowering(std::string intrinsicName);
```

Comment:
I do not think you need to change that one to fix the bug. `getIntrinsicArgumentLowering` is not doing any captures, the change with `name` in ConvertExpr.cpp should be sufficient. Did you change it for a reason ?